### PR TITLE
 [INTERNAL][HTML WB] Added drawing canvas

### PR DESCRIPTION
**Pull request by AbdBarho**
this canvas is used for rendering the shapes on screen, it extends the FabricJs canvas element and inherits all its functionality, it also reacts to changes in window size to fit the window size or fit all elements in it.

all changes to the canvas itself are local (not synchronized).

### CODE REVIEW

File: de.fu_berlin.inf.dpp.whiteboard/frontend/src/components/resizablecanvas.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ * used to calculate the value of each object
+ * @param {Array} array
+ * array of objects
+ */
+function max(valueFunction, array = []) {
+  return Math.max(...array.map(object => valueFunction(object)));
+}
+
+/**
+ * sums the left and width cooridnates to find the furthest point to the right
```
Comment (m273d15) [inline de.fu_berlin.inf.dpp.whiteboard/frontend/src/components/resizablecanvas.js]: typo `coordinates`

File: de.fu_berlin.inf.dpp.whiteboard/frontend/src/components/resizablecanvas.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+ * target object
+ * @return {Number}
+ * the furthest point to the right of the object
+ */
+function rightMostPoint(obj) {
+  return sumAttributes(obj, ["width", "left"]);
+}
+
+/**
+ * sums the top and height cooridnates to find the furthest point to the bottom
```
Comment (m273d15) [inline de.fu_berlin.inf.dpp.whiteboard/frontend/src/components/resizablecanvas.js]: same typo

File: de.fu_berlin.inf.dpp.whiteboard/frontend/src/components/resizablecanvas.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  return sumAttributes(obj, ["height", "top"]);
+}
+
+/**
+ * sums the values of the given attributes in the object
+ * @param {Object} obj
+ * @param {Array} attributes
+ * @return {Number}
+ */
+function sumAttributes(obj = {}, attributes = []) {
```
Comment (m273d15) [inline de.fu_berlin.inf.dpp.whiteboard/frontend/src/components/resizablecanvas.js]: :+1:

File: de.fu_berlin.inf.dpp.whiteboard/frontend/src/components/resizablecanvas.js (hunk 1 above)
Comment (AbdBarho) [reply de.fu_berlin.inf.dpp.whiteboard/frontend/src/components/resizablecanvas.js]: all fixed

**Pull request merged**